voluncheer/profiles/views/profile.py

```diff
@@ -34,4 +38,15 @@ def get_context_data(self, **kwargs):
                 except:
                     pass
             kwargs["badge_urls"] = badge_urls
+            kwargs['user_form'] = VolunteerChangeForm(instance=self.request.user)
         return super().get_context_data(**kwargs)
+
+
+def form_valid(request):
+    """Saves the new user and logs them in."""
+    if request.user.is_volunteer:
+        form = VolunteerChangeForm(request.POST, instance=request.user)
+    elif request.user.is_organization:
+        form = OrganizationChangeForm(request.POST, instance=request.user)
```

Comment:
We should include the `else` block to catch users that do not match those types. e.g. admin.